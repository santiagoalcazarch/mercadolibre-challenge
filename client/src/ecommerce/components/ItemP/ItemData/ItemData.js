import React, { useMemo } from 'react'
import PropTypes from 'prop-types'
import BlueButton from '../../../../shared/components/Buttons/BlueButton'
import { itemCondition as itemConditionEnum } from '../../../../helpers/condition_states'

/**
 * Muestra datos como el valor, el titulo y el estado de 
 * un Item
 */
const ItemData = ({ item }) => {

  const itemCondition = useMemo(() => {
    return itemConditionEnum[item.condition];
  }, [item.condition]);

  const itemPrice = useMemo(() => 
    new Intl.NumberFormat('es-CO', { 
      style: 'currency', 
      currency: item.price?.currency,
      maximumFractionDigits: 0
    }).format( item.price?.amount )
  , [item.price]);

  return (
    <div className="item-l-data">
      <div className="item-l-state-sold"> { itemCondition ? (itemCondition + " - ") : "" } 234 vendidos </div>
      <div className="item-l-name"> { item.title } </div>
      <div className="item-l-price"> { itemPrice } </div>

      <BlueButton
        child="Comprar"
      />
    </div>
  )
}

ItemData.propTypes = {
  item: PropTypes.any.isRequired
}

export default ItemData