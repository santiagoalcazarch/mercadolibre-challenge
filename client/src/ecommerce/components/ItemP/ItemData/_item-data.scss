

.item-l-data{
  width: 100%;
  padding-left: 1rem;

  .item-l-state-sold{
    margin-bottom: 16px;
    font-size: 14px;
  }

  .item-l-name{
    font-size: 24px;
  }

  .item-l-price{
    margin: 32px 0;
    font-size: 46px;
  }
  
}