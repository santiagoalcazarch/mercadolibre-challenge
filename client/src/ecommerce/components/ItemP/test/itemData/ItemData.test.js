
import { shallow } from "enzyme";
import ItemData from "../../ItemData/ItemData";

describe('ItemData component test', () => {

  const item = {
    "id": "MCO3423",
    "title": "iPhone 13", 
    "price": {
      "currency": "COP", 
      "amount": 12000, 
    },
    "picture": "",
    "condition": "new", 
    "free_shipping": true, 
    "sold_quantity": 12,
    "description": "lorem ipsum dolor sit amet, consect"
  }

  const wrapper = shallow( <ItemData item={item} /> );

  test('Debería renderizar el componente correctamente', () => { 
    expect(wrapper).toMatchSnapshot();
  });

  test('Debería mostrar los datos del item', () => { 

    const soldTest = wrapper.find('.item-l-state-sold').text();
    expect(soldTest).toBe("Nuevo - 12 vendidos");

    const numberFormat = new Intl.NumberFormat('es-CO', { 
      style: 'currency', 
      currency: item.price.currency,
      maximumFractionDigits: 0,
      minimumFractionDigits: 0
    }).format( item.price.amount );

    const itemPrice = wrapper.find('.item-l-price').text().trim();
    expect(itemPrice).toBe( numberFormat );

    const itemTitle = wrapper.find('.item-l-name').text().trim();
    expect(itemTitle).toBe(item.title);
  });

  test('Ante un estado de un item desconocido, debería no mostrar ese estado', () => { 

    const itemWithWrongCondition = { ...item }
    itemWithWrongCondition.condition = "anycondition"
    const wrapper = shallow( <ItemData item={itemWithWrongCondition} /> );

    const soldTest = wrapper.find('.item-l-state-sold').text();
    expect(soldTest).toBe("12 vendidos");
  });

})