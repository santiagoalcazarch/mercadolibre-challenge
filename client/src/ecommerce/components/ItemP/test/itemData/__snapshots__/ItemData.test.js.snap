// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemData component test Debería renderizar el componente correctamente 1`] = `
<div
  className="item-l-data"
>
  <div
    className="item-l-state-sold"
  >
    Nuevo - 
    12
     vendidos
  </div>
  <div
    className="item-l-name"
  >
     
    iPhone 13
     
  </div>
  <div
    className="item-l-price"
  >
     
    $ 12.000
     
  </div>
  <BlueButton
    child="Comprar"
  />
</div>
`;
